pyparsing/results.py

```diff
@@ -66,41 +66,41 @@ def test(s, fn=repr):
         - year: 1999
     """
 
+    __slots__ = [
+        "_result__name", "_result__parent", "_result__all_names",
+        "_result__modal", "_result__list", "_result__dict", "__weakref__"
+    ]
+
     def __new__(cls, toklist=None, name=None, asList=True, modal=True):
         if isinstance(toklist, ParseResults):
             return toklist
-        retobj = object.__new__(cls)
-        retobj.__doinit = True
-        return retobj
+        self = object.__new__(cls)
+        self._result__name = None
+        self._result__parent = None
+        self._result__all_names = set()
+        self._result__modal = modal
+        if toklist is None:
+            toklist = []
+        if isinstance(toklist, list):
+            self._result__list = toklist[:]
+        elif isinstance(toklist, _generator_type):
+            self._result__list = list(toklist)
+        else:
+            self._result__list = [toklist]
+        self._result__dict = dict()
+        return self
```

Comment:
The attributes have been changed from their double underscore versions to one with a `_result__*` prefix.  This was done because double-underscore is meant for Python directives, so IDEs do not show them, which makes debugging pyparsing difficult, and also makes debugging `parseActions` in parsers difficult: It is difficult to browse the ParseResult objects.

The longer name has no impact because they are `__slots__` (aka indexes into a tuple)